runtime/src/asset.rs

```diff
@@ -420,6 +423,46 @@ decl_module! {
             Ok(())
         }
 
+        /// Freezes all transfers and minting of a given token.
+        ///
+        /// # Arguments
+        /// * `origin` - the signing key of the sender
+        /// * `ticker` - the ticker of the token
+        pub fn freeze_token(origin, ticker: Ticker) -> DispatchResult {
```

Comment:
And please add `freeze_transfers` & `unfreeze_transfers` functions to the module documentation.

Reply:
Hey @satyamakgec. Please explain what confusion you see. I can't see any. `freeze_transfers` would have been an incomplete name because we disallow minting as well, and minting is not a type of transfer.

How about `freeze_asset` and `unfreeze_asset`, with the corresponding changes in event and storage names?

Reply:
So when we say freeze asset it means we are freezing every functionality of the asset like document addition, checkpoint creation, transfer rights, controller transfers/operations etc.

While `freeze_transfers` only signifies the transfers of the asset are restricted and even `mint` is a special type of transfer where tokens are transferred from 0x0 address to the target address.

Reply:
>where tokens are transferred from 0x0 address to the target address.

This ain't ethereum, we don't do that here xD. I get your point though. FWIW, I am fine with either name.

Reply:
> This ain't ethereum, we don't do that here xD. I get your point though. FWIW, I am fine with either name.

If that's the case then `freeze_transfers` would have been misleading.

Reply:
It looks like we are talking about different definitions of "frozen". According to the requirements, a frozen asset is one with disallowed transfers and minting. The requirements don't mention other functionality which remains allowed. Under this definition, I'd say `freeze_asset` would be OK.

Reply:
> This ain't ethereum, we don't do that here xD. I get your point though. FWIW, I am fine with either name.

Yeah, I agree it is not Ethereum but the gist is same. 

> If that's the case then freeze_transfers would have been misleading.

I don't have any problem with `freeze_asset` name the only thing I want to be sure that each function name represents the same thing that it is doing logically. Sometimes ambiguous names mislead to dApp team. If everyone is ok with `freeze_asset` then we are good to go.